src/backend/distributed/planner/distributed_planner.c

```diff
@@ -527,6 +527,31 @@ GetRTEIdentity(RangeTblEntry *rte)
 }
 
 
+/*
+ * GetQueryLockMode returns the necessary lock mode to be acquired for given
+ * query according to "rellockmode" field of RangeTblEntry struct from
+ * parsenodes.h in PostgreSQL 12.x.
+ */
+LOCKMODE
+GetQueryLockMode(Query *query)
```

Comment:
Why isn't there an `#ifdef` if this is some version specific thing?

Reply:
Actually the case is, if we want to get necessary lockmode for a relation (`RangeVar`) within a query, we can get the lock mode easily from the `RangeVar` itself.

However, if we want to decide to the lockmode appropriate for the "query", we do the above in both versions.

Reply:
(Pr title & description should be changed)